src/Apps/Order/Components/AddressVerificationFlow.tsx

```diff
@@ -0,0 +1,309 @@
+import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
+import { createFragmentContainer, graphql } from "react-relay"
+import { AddressVerificationFlowQuery } from "__generated__/AddressVerificationFlowQuery.graphql"
+import { AddressVerificationFlow_verifiedAddressResult$data } from "__generated__/AddressVerificationFlow_verifiedAddressResult.graphql"
+import {
+  BorderedRadio,
+  Box,
+  Button,
+  Flex,
+  ModalDialog,
+  RadioGroup,
+  Spacer,
+  Text,
+} from "@artsy/palette"
+import { useCallback, useEffect, useState } from "react"
+
+interface AddressVerificationFlowProps {
+  verifiedAddressResult: AddressVerificationFlow_verifiedAddressResult$data
+  onChosenAddress: (address: AddressValues) => void
+  onClose: () => void
+}
+
+export interface AddressValues {
+  addressLine1: string
+  addressLine2?: string
+  country: string
+  city: string
+  region: string
+  postalCode: string
+}
+
+enum ModalType {
+  SUGGESTIONS = "suggestions",
+  REVIEW_AND_CONFIRM = "review_and_confirm",
+}
+
+interface AddressOption {
+  key: string
+  recommended: boolean
+  lines: string[]
+  address: AddressValues
+}
+
+enum AddressSuggestionRadioButton {
+  recommended = "Recommended",
+  user_address = "What you entered",
+}
+
+const AddressVerificationFlow: React.FC<AddressVerificationFlowProps> = ({
+  verifiedAddressResult,
+  onChosenAddress,
+  onClose,
+}) => {
+  console.log({ queryResult: verifiedAddressResult })
+  const [modalType, setModalType2] = useState<ModalType | null>(null)
+  const [addressOptions, setAddressOptions2] = useState<AddressOption[]>([])
+
+  const [selectedAddressKey, setSelectedAddressKey2] = useState<string | null>(
+    null
+  )
+
+  const setModalType = (type: ModalType | null) => {
+    console.log("modal", type)
+    setModalType2(type)
+  }
+  const setAddressOptions = (options: AddressOption[]) => {
+    console.log("options", options)
+    setAddressOptions2(options)
+  }
+  const setSelectedAddressKey = (key: string) => {
+    console.log("key", key)
+    setSelectedAddressKey2(key)
+  }
+
+  const chooseAddress = useCallback(() => {
+    if (!selectedAddressKey) return
+
+    const selectedAddress = addressOptions.find(
+      option => option.key === selectedAddressKey
+    )
+    if (selectedAddress) {
+      setModalType(null)
+      onChosenAddress(selectedAddress.address)
+    }
+  }, [addressOptions, onChosenAddress, selectedAddressKey])
+
+  const onClose2 = useCallback(() => {
+    setModalType(null)
+    onClose()
+  }, [onClose])
+
+  useEffect(() => {
+    if (addressOptions.length > 0) {
+      setSelectedAddressKey(addressOptions[0].key)
+    }
+  }, [addressOptions])
+
+  // Handling possibly incorrectly (by MP) possible null values
```

Comment:
I would just fix the types in Metaphysics

Reply:
yes, i agree. i'll make a ticket.